### PR TITLE
Fix for Issue833

### PR DESCRIPTION
**Pull request by amidzak**
This pull fixes the issue of double implements keywords for queued and pooled state machines when the containing class implements an interface.

### CODE REVIEW

File: UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump [hunk 1]
```diff
@@ -98,17 +98,27 @@ public <<# if (uClass.getIsAbstract()) { append(realSb, "{0} ", "abstract"); } #
 for (StateMachine smq : uClass.getStateMachines())
   {
     if (smq.isQueued())
-    {
-      append(realSb," implements Runnable");
+    { 
+      if (uClass.hasParentInterface() == false){
```
Comment (Nava2) [inline UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump:102]: should probably just be `if (!uClass.hasParentInterface())`

Comment (Nava2) [inline UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump:102]: Is there ever a case that the model has more than a single interface to implement? or it always going to just be `Runnable`?

Comment (amidzak) [inline UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump:102]: @Nava2  The problem here is the implementation of code generation for queued and pooled state machines. The implemented class needs to implement Runnable. For other interfaces which are in the model, there is another method that lists all of them.

Comment (vahdat-ab) [inline UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump:102]: I just remember I have seen the case in which in the code generator we had two `implements` keywords which would make we cannot compile the java code. Now, I cannot get the example from the top of my head, but if now you're dealing with this case, you may want to check it as well.

Comment (vahdat-ab) [inline UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump:102]: @amindzak Please check the issue #840.
You may be able to fix it easily in this part of the code.

Comment (amidzak) [inline UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump:102]: @vahdat-ab issue #840 is the same with issue 833. This pr fixes that. When I build on my computer it doesn't fail. Do you know what would be the problem? it doesn't pass AppVeyor.

Comment (vahdat-ab) [inline UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump:102]: @amidzak sorry about that. I completely forgot we talked about it before. 
The issue is not related to your PR. You are good to go.

Comment (vahdat-ab) [inline UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump:102]: I wonder if you can merge it or not?
If not, I checked your PR, It doesn't show that wrong behavior and I can merge it.

**Pull request merged**